File: app/src/main/java/com/dashboard/TransportSelectionFragment.java
package com.dashboard

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.dashboard.databinding.FragmentTransportSelectionBinding

class TransportSelectionFragment : Fragment() {

    private var _binding: FragmentTransportSelectionBinding? = null
    private val binding get() = _binding!!

            override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View {
        _binding = FragmentTransportSelectionBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.btnPlane.setOnClickListener {
            Toast.makeText(requireContext(), "Plane selected", Toast.LENGTH_SHORT).show()
            // findNavController().navigate(R.id.action_transportSelectionFragment_to_airlineSelectionFragment)
        }

        binding.btnShip.setOnClickListener {
            Toast.makeText(requireContext(), "Ship selected", Toast.LENGTH_SHORT).show()
            // findNavController().navigate(R.id.action_transportSelectionFragment_to_shipSelectionFragment)
        }

        binding.btnCar.setOnClickListener {
            Toast.makeText(requireContext(), "Car selected", Toast.LENGTH_SHORT).show()
            // findNavController().navigate(R.id.action_transportSelectionFragment_to_carSelectionFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}

File: app/src/main/java/com/android/tripbook/ViewModel/TransportSelectionFragment.kt
package com.android.tripbook.ViewModel

m.yourpackage.R
import com.google.android.material.button.MaterialButton

class TransportSelectionFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_transport_selection, container, false)

        val planeBtn = view.findViewById<MaterialButton>(R.id.btnPlane)
        val shipBtn = view.findViewById<MaterialButton>(R.id.btnShip)
        val carBtn = view.findViewById<MaterialButton>(R.id.btnCar)

        planeBtn.setOnClickListener {
            // Navigate to airline options fragment
        }

        shipBtn.setOnClickListener {
            // Navigate to ship options fragment
        }

        carBtn.setOnClickListener {
            // Navigate to car options fragment
        }

        return view
    }
}